airbyte-integrations/connectors/destination-redis/src/main/java/io/airbyte/integrations/destination/redis/RedisSslUtil.java

```diff
@@ -0,0 +1,155 @@
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RedisSslUtil {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(RedisSslUtil.class);
+
+  public static final String PARAM_CLIENT_CERTIFICATE = "client_certificate";
+  public static final String PARAM_CLIENT_KEY = "client_key";
+  public static final String PARAM_CA_CERTIFICATE = "ca_certificate";
+  public static final String PARAM_CLIENT_KEY_PASSWORD = "client_key_password";
+  public static final String PARAM_SSL_MODE = "mode";
+  public static final String PARAM_SSL_MODE_VERIFY_FULL = "verify-full";
+  public static final String PARAM_SSL = "ssl";
+
+  private static final String CLIENT_CERTIFICATE = "client.crt";
+  private static final String CLIENT_CA_CERTIFICATE = "client-ca.crt";
+  private static final String CLIENT_KEY = "client.key";
+  private static final String CLIENT_KEY_STORE = "client_key_store.p12";
+  private static final String KEY_STORE_TYPE = "PKCS12";
+  private static final String TRUST_STORE = "truststore.jks";
+  private static final String TRUST_PASSWORD = "truststore_pwd";
+  private static final String TRUST_TYPE = "JKS";
+
+  /**
+   * set javax.net.ssl.keyStore and javax.net.ssl.trustStore based on provided ca.crt, client.crt, client.kay
+   *
+   * @param redisSslMode json ssl mode config
+   */
+  public static void setupCertificates(final JsonNode redisSslMode) {
+    try {
+      if (getSslVerifyMode(redisSslMode) == VERIFY_IDENTITY) {
+        LOGGER.info("Preparing ssl certificates for {} mode", PARAM_SSL_MODE_VERIFY_FULL);
+        final String clientKeyPassword = getOrGeneratePassword(redisSslMode);
+        initCertificateStores(redisSslMode.get(PARAM_CA_CERTIFICATE).asText(),
+            redisSslMode.get(PARAM_CLIENT_CERTIFICATE).asText(), redisSslMode.get(PARAM_CLIENT_KEY).asText(), clientKeyPassword);
+      }
+    } catch (final IOException | InterruptedException e) {
+      throw new RuntimeException("Failed to import certificate into Java Keystore");
+    }
+  }
+
+  /**
+   * Generate random pass if key pass param is empty
+   *
+   * @param sslModeConfig json ssl mode config
+   * @return client key password
+   */
+  private static String getOrGeneratePassword(final JsonNode sslModeConfig) {
+    final String clientKeyPassword;
+    if (sslModeConfig.has(PARAM_CLIENT_KEY_PASSWORD) && !sslModeConfig.get(PARAM_CLIENT_KEY_PASSWORD).asText().isEmpty()) {
+      clientKeyPassword = sslModeConfig.get(PARAM_CLIENT_KEY_PASSWORD).asText();
+    } else {
+      clientKeyPassword = RandomStringUtils.randomAlphanumeric(10);
+    }
+    return clientKeyPassword;
+  }
+
+  /**
+   * The method generate certificates based on provided ca.crt, client.crt, client.kay.
```

Comment:
think this is supposed to be `client.key`, right?

Reply:
Thanks!